Core.ApplicationServices/Contract/Write/ItContractWriteService.cs

```diff
@@ -270,6 +274,69 @@ private Maybe<OperationError> UpdateDataProcessingRegistrations(ItContract contr
             );
         }
 
+        private Result<ItContract, OperationError> UpdatePaymentModelParameters(ItContract contract, ItContractPaymentModelModificationParameters parameters)
+        {
+            return contract
+                .WithOptionalUpdate(parameters.OperationsRemunerationStartedAt, (c, newValue) => c.OperationRemunerationBegun = newValue.HasValue ? newValue.Value : null)
+                .Bind(itContract => itContract.WithOptionalUpdate(parameters.PaymentFrequencyUuid, UpdatePaymentFrequency))
+                .Bind(itContract => itContract.WithOptionalUpdate(parameters.PaymentModelUuid, UpdatePaymentModel))
+                .Bind(itContract => itContract.WithOptionalUpdate(parameters.PriceRegulationUuid, UpdatePriceRegulation))
+                .Bind(itContract => itContract.WithOptionalUpdate(parameters.PaymentMileStones, UpdatePaymentMileStones));
+        }
+
+        private Maybe<OperationError> UpdatePaymentMileStones(ItContract contract, IEnumerable<ItContractPaymentMilestone> milestones)
+        {
+            //Replace existing trials (duplicates are allowed so we cannot derive any meaningful unique identity)
+            var paymentMilestones = contract.PaymentMilestones.ToList();
+            paymentMilestones.ForEach(_paymentMilestoneRepository.Delete);
+            contract.ResetPaymentMilestones();
```

Comment:
for den logiske læsers skyld ville det måske være rart om du kaldte "reset" inden du kalder Delete :-)